### PR TITLE
Add docs about package verification

### LINKED ISSUE #2005: [Request][Fleet] Package signatures documentation
## Description

As part of https://github.com/elastic/kibana/issues/133822 we are adding package signature verification to Kibana. 

This is the same signing that we use for our code releases. e.g elastic binaries. Package signatures prevent attackers from modifying package content, when a package is downloaded, Kibana verifies the packages signature against Elastics public key meaning the package content can be truested. 

If a user tries to install a package which fails verification, they are presented with a modal which gives them the option to force install.

The original issue has some more details about why package signing is useful https://github.com/elastic/package-spec/issues/46

We need some content for the learn more link from the designs:

https://user-images.githubusercontent.com/6766512/172449124-5c50b54b-bb23-4de6-87a1-bfc69915743b.png
https://user-images.githubusercontent.com/6766512/172449239-c6d7e30a-5e3e-4fc6-b19a-2e74675782d2.png

## Collaboration
- The docs team will define with product team the structure and location, and the product team will provide the initial content

I am happy to have a stab at writing the document but I could maybe do with a chat with someone about the ideal structure/content

**Contact Person:** @hop-dev 

## Suggested Target Release

8.4

### PR DESCRIPTION
**Pull request by dedemorton**
Closes #2005 

Please review. Some of the UI-related details are fuzzy because the feature is not yet turned on in the build.

Remaining questions/issues:

- [x] @kpollich Which versions (if any) should I backport this to?
- [x] @elastic/ecosystem Will the docker image for the air gapped registry will have signatures?
- [x] @hop-dev Who is updating the docs at https://www.elastic.co/guide/en/kibana/master/fleet-settings-kb.html to show the package verification setting?
- [x] @dedemorton Need to verify this info by looking at the UI when the feature is turned on. (See if this is necessary.)
- [x] It sounds like we are blocked and shouldn't merge this until https://github.com/elastic/ingest-dev/issues/1193 is resolved.

### CODE REVIEW

File: docs/en/ingest-management/integrations/package-signatures.asciidoc
Comment (dedemorton) [inline docs/en/ingest-management/integrations/package-signatures.asciidoc:48]:
```suggestion
- Does this change affect the https://www.elastic.co/guide/en/fleet/current/air-gapped.html[air gapped docs]?
```

File: docs/en/ingest-management/integrations/package-signatures.asciidoc
Comment (hop-dev) [inline docs/en/ingest-management/integrations/package-signatures.asciidoc:31]: > What if a key has to be invalidated in the future?

If the elastic signing key changed, then existing package installs would still show as verified (as they were verified at the time of install) but they would then show as unverified when new packages are installed or existing ones are upgraded. You are right that the user can update the ocnfig (see answer below) to make this work again. 

Some more info on this here: https://github.com/elastic/elastic-package/issues/576#issuecomment-1197867649

> Would users provide the file path in kibana.yml (through the xpack.fleet.packageVerification.gpgKeyPath setting)?

Yes this is what they would have to do for existing deployments, the key would be updated on kibana upgrade as well.

>  Do we need to worry about documenting this now?

Maybe a small nod to the `xpack.fleet.packageVerification.gpgKeyPath` setting? Not sure!

File: docs/en/ingest-management/integrations/package-signatures.asciidoc
Comment (hop-dev) [inline docs/en/ingest-management/integrations/package-signatures.asciidoc:43]: - Can anyone who has access to the integrations UI install unverified packages?

Currently yes

File: docs/en/ingest-management/integrations/package-signatures.asciidoc
Comment (hop-dev) [inline docs/en/ingest-management/integrations/package-signatures.asciidoc:47]: The docker image will have signatures when it is updated I believe @mtojek? the only complication is that kibana supports verification from 8.4 onwards but the registry may only have signatures from a later version? So if this doc went into 8.4 the 8.4 registry docker image wouldn't provide signatures and verification wouldnt be performed, but it wouldnt break anything

Comment (dedemorton) [reply docs/en/ingest-management/integrations/package-signatures.asciidoc:83]: @mtojek Can you let me know if the docker image for the air gapped registry will have signatures? I need to know if the air gapped docs need to be updated, too.

**Comment by dedemorton** [comment]
I've marked this PR as ready for review because I've taken it as far as I can without seeing the UI in a build. @hop-dev Are we still planning to turn this on for 8.5?

**Comment by dedemorton** [comment]
@hop-dev Has this feature made it into 8.5?  I need someone to review the content.

@mtojek Can you please respond to the question about whether the air gapped registry will have signatures?

**Comment by mrodm** [comment]
> @mtojek Can you please respond to the question about whether the air gapped registry will have signatures?

Currently, the docker images that are built for air-gapped envs and used in those Elastic Package registries do not contain the signatures. But we plan to include them soon. This change should be done after switching the DNS to start using package storage v2.

cc @jsoriano

**Comment by dedemorton** [comment]
Added "blocked" label because I am blocked until https://github.com/elastic/ingest-dev/issues/1193 is merged.

@jen-huang While we're waiting for the DNS switch to v2, can someone please review the content here (and answer my questions) so that this content is ready to go when the feature goes live? Some insight into current plans would also be useful. Thanks!

**Comment by jlind23** [comment]
@hop-dev @kpollich could you please review this content as asked by @dedemorton ?

**Review by kpollich — APPROVED** [review]
Thanks for this - docs look good. I have filed a separate PR updating the Fleet settings docs, and if the airgapped docs need an update I think it'd be fine to file a followup.

File: docs/en/ingest-management/integrations/package-signatures.asciidoc
Comment (mrodm) [reply docs/en/ingest-management/integrations/package-signatures.asciidoc:83]: > @jsoriano @mrodm - will the `docker.elastic.co/package-registry/distribution:8.6.0` contain `signature_path` values in these responses?

@kpollich  Yes, with the new Package Storage V2 being launched, Jenkins jobs have been updated to generate the docker distribution images with the signatures. Thereby, the following docker distribution images will have the `signature_path` field for all packages.

Just tried locally one of the latest production images built and pushed in Jenkins and that field is already present:

```shell
 $ curl -s "http://localhost:8080/search?package=system" |jq -r '.[] | ."signature_path"'
/epr/system/system-1.20.4.zip.sig
```

**Comment by dedemorton** [comment]
@kpollich A couple of quick follow-up questions before I merge this:
* Which versions (if any) should I backport this to?
* Is there an easy way to install an unverified package to confirm the details here? Or maybe that's not necessary since you have reviewed the info?

**Comment by kpollich** [comment]
> Which versions (if any) should I backport this to?

We can backport this to 8.5.

> Is there an easy way to install an unverified package to confirm the details here? Or maybe that's not necessary since you have reviewed the info?

I don't think this is necessary. We ran through these scenarios in our testing/development cycle and everything documented here looks accurate. @hop-dev's previous review covered any of the functionality specifics, etc so I think we're good here.

**Pull request merged**
